### ISSUE TITLE
Dvorak doesn't work: order of rules chooses us(dvorak) instead of planet_vndr/cosmo(dvorak)

### ISSUE THREAD

**Issue by cgevans** [gemian/xkeyboard-config#8]
In evdev / base rules, `*  dvorak = pc+us(dvorak)` comes before `planetcosmo * = pc+planet_vndr/cosmo(%l)`.  As a result, trying to set the keyboard map to model planetcosmo and layout dvorak chooses the non-planetcosmo dvorak layout, which results in there being no s key, amongst other problems.

I can fix this by reordering the lines, but I'm not quite sure how to do so in the "correct" way, by changing the .part files, and so have this as an issue rather than directly making a PR.

**Comment by adamboardman** [comment]
That's really awkward as you cannot even login given the default password.
I just went with renaming our layout to dvorak-us as the layout has the # symbol as shift-3.

**Comment by cgevans** [comment]
An alternative solution I'm playing with that would preserve the layout name would be to add the model-specific lines to `rules/compat/base.ml_s.part`: I had tried changing the order of the standard and compatibility rules in Makefile.am, which would also fix the problem, but that involves a more substantial change to the rules files that might make it more difficult to get these changes incorporated upstream.

**Comment by adamboardman** [comment]
Interesting, technically the layout is dvorak-us, as in I'd imagine UK folks would actually prefer having the £ a little more easily available. Part of the reason we've got things the way they are is to allow the plugging in of a usb keyboard (or bt) and it using its own keyboard map, tapping on either keyboard re-aligns the shortcuts back to that keyboard. Its not something I've actually tested on the Cosmo but I did with the Gemini, so if your solution would allow that to work well then I'd be all for it. I'll reopen to see if you figure something out something better.
